src/js/components/FilterInputText.js

```diff
@@ -1,121 +1,130 @@
-var classNames = require('classnames');
-var React = require('react');
+import classNames from 'classnames';
+import React, {PropTypes} from 'react';
 
 import Icon from './Icon';
 import ServiceFilterTypes from '../constants/ServiceFilterTypes';
 
-var FilterInputText = React.createClass({
+let METHODS_TO_BIND = [
+  'handleBlur',
+  'handleFocus',
+  'handleChange',
+  'handleInputClear'
+];
 
-  displayName: 'FilterInputText',
+class FilterInputText extends React.Component {
+  constructor() {
+    super(...arguments);
 
-  propTypes: {
-    handleFilterChange: React.PropTypes.func.isRequired,
-    inverseStyle: React.PropTypes.bool,
-    placeholder: React.PropTypes.string,
-    searchString: React.PropTypes.string.isRequired,
-    sideText: React.PropTypes.node
-  },
-
-  getDefaultProps: function () {
-    return {
-      inverseStyle: false,
-      placeholder: 'Filter',
-      searchString: ''
-    };
-  },
-
-  getInitialState: function () {
-    return {
+    this.state = {
       focus: false
     };
-  },
 
-  componentDidUpdate: function () {
-    if (this.state.focus) {
-      this.refs.filterInput.focus();
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.focus !== this.state.focus && this.state.focus && this.inputField) {
+      this.inputField.focus();
     }
-  },
+  }
+
+  handleChange(e) {
```

Comment:
#nit this could be in one line:
```js
  handleChange(event) {
    this.props.handleFilterChange(event.target.value, ServiceFilterTypes.TEXT);
  }
```